File: simulator(js)/legitimate_traffic.js



// # this file will be used to have a notion of legitimate traffic
// # how much legtimate traffic is received at any given time
// # what is the latency observed and all those things will be modeled here
const packet = require("./packet")
const network = require("./network")

const EventEmitter = require('events');

class flowGen extends EventEmitter {}

const flowGen = new flowGen();

let i = 0

flowGen.on('send', (pkt,n) => {
  console.log('packet received to send on network!');
  network.sendtoNetwork(pkt);
  i++;
  if(i < n) {
  	self.emit("send")
  }
});


flowGen.emit('send',pkt,n);

// def FlowGenSimple(pkt_no):
//   for i in range(1,pkt_no):
//     sendtoNetwork(pkt)
  

File: jython/legitimate_traffic.py



# this file will be used to have a notion of legitimate traffic
# how much legtimate traffic is received at any given time
# what is the latency observed and all those things will be modeled here
import packet
import globals
import time
import network
import logging

def flowGen():
	# globals.DEBUG_LOGGER.debug("Function: flowGen")
	if(globals.LEG_TRAFFIC_MODEL == "simple"):
		flowGenSimple()

def sendPkts(n):
	for i in range(0,n):
		network.sendtoNetwork(packet.Packet(globals.PKT_LEN,"udp",0,0))

def flowGenSimple():
	fixedRate = 1 # Mbps
	numPkts = int(fixedRate*1.0 / (globals.PKT_LEN + 1))
	globals.DEBUG_LOGGER.debug("Function: flowGenSimple - Number of packets to send in 1 second %s",str(numPkts))
	while True:
		sendPkts(numPkts)
		time.sleep(1)
		# network.sendtoNetwork(pkt)





# def randFlowGen():
  
